Other tests

IdanTo1 opened the pull request.

IdanTo1 committed:
implement Future and Inventory tests

Signed-off-by: Idan Tomer <idan.tomer2@gmail.com>

IdanTo1 committed:
Merge remote-tracking branch 'origin/master' into otherTests

IdanTo1 committed:
1. add constructor to Agent (won't compile without)
2. add squad tests

Signed-off-by: Idan Tomer <idan.tomer2@gmail.com>

IdanTo1 committed:
Merge remote-tracking branch 'origin/master' into otherTests

IdanTo1 committed:
update squad to be static, to load it only once

Signed-off-by: Idan Tomer <idan.tomer2@gmail.com>

Hroy02 commented on src/main/java/bgu/spl/mics/application/passiveObjects/Agent.java:
```diff
@@ -8,6 +8,9 @@
  */
 public class Agent {
 
+	public Agent(String s, String a) {
+	}
+
```
Perhaps use more informative naming

IdanTo1 replied:
I did it like this just because we're not submitting it, and we'll do it appropriately in future when needed

Hroy02 replied:
Oh right this isn't getting submitted. Cool :)

Hroy02 commented on src/test/java/bgu/spl/mics/FutureTest.java:
```diff
@@ -2,17 +2,30 @@
+        assertTrue(f.isDone() == false, "unresolved future marked as done");
+        f.resolve(TEST_STRING);
+        assertTrue(f.get(100, t) == TEST_STRING, "unresolved future differs from null");
+        assertTrue(f.get() == TEST_STRING, "unresolved future differs from null");
```
The message is wrong. It should be "resolved future doesn't contain proper result" or something similar.

Hroy02 commented on src/test/java/bgu/spl/mics/InventoryTest.java:
```diff
@@ -1,19 +1,50 @@
+
+    @Test
+    public void testSingleTon(){
+        assertTrue(inv != null, "singleTon not created");
```
Singleton is an English word, there is no need to capitalize the T.

IdanTo1 replied:
is it? :laughing:

Hroy02 replied:
Yeah, actually it's the direct translation of the word "יחידון".
[Link to Merriam-Webster definition of the word](https://www.merriam-webster.com/dictionary/singleton)

Hroy02 commented on src/test/java/bgu/spl/mics/InventoryTest.java:
```diff
@@ -1,19 +1,50 @@
+            inv.getItem(gadget); // flush inventory
+        }
+        String[] tmp = {};
+        inv.load(tmp); //TODO complete read to string array json
```
Still needs to be done

Hroy02 commented on src/test/java/bgu/spl/mics/SquadTest.java:
```diff
@@ -1,19 +1,62 @@
+
+    @Test
+    public void testSingleTon(){
+        assertNotNull(squad, "singleTon not created");
```
Same singleton thing

Hroy02 commented on src/test/java/bgu/spl/mics/SquadTest.java:
```diff
@@ -1,19 +1,62 @@
+        for (Agent agent: agents) {
+            testSerials.add(agent.getSerialNumber());
+            testNames.add(agent.getName());
+        }
```
This isn't getting the agents you loaded to the Squad. It takes them from the list of agents you created in the beginning. This is maybe a test the agents weren't changed. But not a test for `load()`

IdanTo1 replied:
I'm checking that all serials in the squad has the appropriate name and vice versa,  and so I check existence and validity.

IdanTo1 replied:
no other way to do this because agents in squad is private..

IdanTo1 replied:
no other way to do this because agents in squad is private..

Hroy02 replied:
But Agents is just the arraylist you created. It makes no sense.
On the other hand by using getAgents(serials) you would get the agents stored in the Squad object. Which is what you need.

IdanTo1 replied:
Ohhh ok
I was sure this what I do. Now I understand

Hroy02 reviewed (changes requested):
Mostly grammar which I'm picky about.
The one problem I saw was in squad which I think isn't doing what we intended

IdanTo1 committed:
1. FutureTest - fix fail message
2. InventoryTest - remove printToFile test and fix singleton write
3. SquadTest - fix singleton write
4. general improvements

Signed-off-by: Idan Tomer <idan.tomer2@gmail.com>

IdanTo1 committed:
remove redundant for loop creating and existing array.
because squad doesn't have getSerials method, this is the only thing we can check.
In this test we are trusting getAgentsNames to work

implement getInstance to resolve a warning

Signed-off-by: Idan Tomer <idan.tomer2@gmail.com>

Hroy02 reviewed (approved):
Great work!
That's exactly what I meant.
